### PR TITLE
[Doc] Explain programming exercise options

### PR DESCRIPTION
**Pull request by kloessst**
### Motivation and Context

Explain the various options for the creation of programming exercises
https://artemis-platform.readthedocs.io/en/docs-programming-exercise-options/user/exercises/programming/#exercise-creation

### CODE REVIEW

File: docs/user/exercises/programming-exercise-setup.rst [hunk 1]
```diff
@@ -87,14 +90,125 @@ Exercise Creation
...
+      the exercise creation.
+    |
+
+    .. figure:: programming/programming-options-programming-language.png
+              :align: center
+
+    - **Programming Language:** The programming language for the exercise. Artemis chooses the template accordingly.
+      Refer to the :ref:`programming exercise features <programming_features>` for an overview of the supported features for each template.
+    - **Project Type:** Determines the project structure of the template. Not available for all programming languages.
+    - **Package Name:** The package name used for this exercise. Not available for all programming languages. Not available for all programming languages.
```
Comment (FuchsDominik) [inline docs/user/exercises/programming-exercise-setup.rst:129]:
```suggestion
    - **Package Name:** The package name used for this exercise. Not available for all programming languages.
```

File: docs/user/exercises/programming-exercise-setup.rst [hunk 2]
```diff
@@ -87,14 +90,125 @@ Exercise Creation
...
+    |
+
+    .. figure:: programming/programming-options-timeline.png
+              :align: center
+
+    - **Release Date:** Release date of the exercise. Students will only be able to participate in the exercise after this date.
+    - **Automatic Tests:** Every commit of a participant triggers the execution of the tests in the **Test** repository.
+      Exempt are tests, which are specified to run after the due date. This is only possible if **Run Tests once after Due Date:** has been activated.
```
Comment (FuchsDominik) [inline docs/user/exercises/programming-exercise-setup.rst:137]:
```suggestion
      Excluded are tests, which are specified to run after the due date. This is only possible if **Run Tests once after Due Date:** has been activated.
```

Comment (FuchsDominik) [reply docs/user/exercises/programming-exercise-setup.rst:137]: Probably what you meant?

File: docs/user/exercises/programming-exercise-setup.rst [hunk 3]
```diff
@@ -87,14 +90,125 @@ Exercise Creation
...
+    - **Project Type:** Determines the project structure of the template. Not available for all programming languages.
+    - **Package Name:** The package name used for this exercise. Not available for all programming languages.
+    |
+
+    .. figure:: programming/programming-options-timeline.png
+              :align: center
+
+    - **Release Date:** Release date of the exercise. Students will only be able to participate in the exercise after this date.
+    - **Automatic Tests:** Every commit of a participant triggers the execution of the tests in the **Test** repository.
+      Excluded are tests, which are specified to run after the due date. This is only possible if **Run Tests once after Due Date:** has been activated.
```
Comment (kloessst) [inline docs/user/exercises/programming-exercise-setup.rst:137]:
```suggestion
      Excluded are tests, which are specified to run after the due date. This is only possible if **Run Tests once after Due Date** has been activated.
```

**Review by krusche — APPROVED** [review]
Looks good as another iteration on the way to a better documentation :-)

**Pull request merged**